actionview/lib/action_view/helpers/asset_tag_helper.rb

```diff
@@ -364,6 +375,16 @@ def multiple_sources_tag_builder(type, sources)
           end
         end
 
+        def resolve_image_source(source, skip_pipeline)
+          if source.is_a?(Symbol) || source.is_a?(String)
+            path_to_image(source, skip_pipeline: skip_pipeline)
+          else
+            polymorphic_url(source)
```

Comment:
`url_for` seems to truncate absolute URLs the same way `polymorphic_path` does (and `polymorhic_url` doesn't). That is, `resolve('Attachment') { "http://dropbox.com/box456" }` would be turned into `/box456`.